specs/Print.md

```diff
@@ -0,0 +1,473 @@
+                    return S_OK;
+                })
+                .Get()));
+     return true;
+}
+```
+
+```c#
+async void PrintToPdfStream()
+{
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings();
+    printSettings.Orientation = CoreWebView2PrintOrientation.Portrait;
+    printSettings.ShouldPrintBackgrounds = true;
+
+    MemoryStream pdfStream = new MemoryStream();
+    System.IO.Stream stream = await webView.CoreWebView2.PrintToPdfStreamAsync(printSettings);
+    DisplayPdfDataInPrintDialog(pdfData);
+}
+
+// Function to display current page pdf data in a custom print preview dialog.
+void DisplayPdfDataInPrintDialog(Stream pdfData)
+{
+    // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+/// Specifies the print dialog kind.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_DIALOG_KIND {
+  /// Opens the browser print preview dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW,
+
+  /// Opens the system print dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_SYSTEM_PRINT,
+
+} COREWEBVIEW2_PRINT_DIALOG_KIND;
+
+/// Specifies the printing side.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_SIDE {
+  /// Single-sided printing.
+  COREWEBVIEW2_PRINT_SIDE_SINGLE,
+
+  /// Double-sided, horizontal printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_HORIZONTAL,
+
+  /// Double-sided, vertical printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_VERTICAL,
+
+} COREWEBVIEW2_PRINT_SIDE;
+
+[uuid(92269C88-1CC1-47E0-9A87-71458409BA15), object, pointer_default(unique)]
+interface ICoreWebView2_15 : ICoreWebView2_14 {
+  /// Print the current page asynchronously to the specified printer with the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. The handler will return `E_ABORT`
+  /// if `printerName` doesn't match with the name of any printers on the OS
+  /// or invalid settings for a given printer.
+  ///
+  /// The async `PrintWithSettings` operation completes when it finishes
+  /// printing to the printer. At this time the `ICoreWebView2PrintWithSettingsCompletedHandler`
+  /// is invoked. Only one `Printing` operation can be in progress at a time. If
+  /// `PrintWithSettings` is called while a `PrintWithSettings` or `PrintToPdf` or `PrintToPdfStream` or `Print`
+  /// job is in progress, the completed handler is immediately invoked with an error.
+  ///
+  /// \snippet AppWindow.cpp PrintWithSettings
+  HRESULT PrintWithSettings(
+    [in] ICoreWebView2PrintSettings2* printSettings,
+    [in] ICoreWebView2PrintWithSettingsCompletedHandler* handler);
+
+  /// Opens the print dialog to print the current page. See `COREWEBVIEW2_PRINT_DIALOG_KIND`
+  /// for descriptions of print dialog kinds.
+  /// The default value is `COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW`.
+  ///
+  /// \snippet AppWindow::OpenPrintDialog
```

Comment:
What happens if the page navigates while the Print dialog is displayed (but the user hasn't clicked "Print" yet)? Is page navigation paused until the dialog is dismissed? Does page navigation dismiss the dialog? Does the dialog stay up, but it still prints the old page when the user clicks Print? Or does it print the new page? Or does it just error out?

Similarly, what happens if the page navigates while it is printing?

Reply:
If the page navigates when a print dialog is displayed, the dialog is dismissed, and navigation continues.

If the page navigates while it is printing (upon print click), page continues to navigate and old page gets printed.